drivers/serial/uart_mcux_lpuart.c

```diff
@@ -456,3 +476,37 @@ static const struct mcux_lpuart_config mcux_lpuart_##n##_config = {	\
 	LPUART_MCUX_INIT_CFG(n);
 
 DT_INST_FOREACH_STATUS_OKAY(LPUART_MCUX_INIT)
+#else
+
+#define LPUART_MCUX_DECLARE_CFG(n, IRQ_FUNC_INIT)			\
+static const struct mcux_lpuart_config mcux_lpuart_##n##_config = {	\
+	.base = (LPUART_Type *) DT_INST_REG_ADDR(n),			\
+	.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),		\
+	.clock_subsys = (clock_control_subsys_t)DT_INST_CLOCKS_CELL(n, name),\
+	.baud_rate = DT_INST_PROP(n, current_speed),			\
+	.flow_ctrl = DT_INST_PROP(n, hw_flow_control) ?	\
+		UART_CFG_FLOW_CTRL_RTS_CTS : UART_CFG_FLOW_CTRL_NONE,\
+	IRQ_FUNC_INIT							\
+}
```

Comment:
That's the right solution, I believe I went with this option initially because the existing definitions were already duplicated. I've cleaned up the device definitions in the LPUART driver, and the latest push uses conditional sub-defines